internal/model/rwfolder.go

```diff
@@ -644,7 +644,16 @@ func (p *rwFolder) handleDir(file protocol.FileInfo) {
 			if err != nil || p.ignorePermissions(file) {
 				return err
 			}
-			return os.Chmod(path, mode)
+
+			// Stat the directory so we can check its permissions.
+			info, err := osutil.Lstat(path)
+			if err != nil {
+				return err
```

Comment:
Yes exactly. I think this whole bug is here because `os.Chmod` sets the extra filemode bits to 0 where as linux chmod does not.